satellite/marketing/marketingweb/server_test.go

```diff
@@ -69,6 +72,11 @@ func TestCreateOffer(t *testing.T) {
 					return err
 				}
 
+				_, err = http.Get(baseURL + "/stop/" + id)
```

Comment:
Should it be a `PUT` request if we are updating the data ?

Reply:
Also, if this is working, then we are missing the appropriate method check.

Reply:
I'm not using a `PUT` request I'm using a `GET` and generating dynamic params
`<a href="/stop/{{range .FreeCredits}}{{if eq .Status 2}}{{.ID}}{{end}}{{end}}">`

There's no need to use a form for this and even if we did you can't use `PUT` methods in an html form.

Reply:
If this is doing a modification then it must not use GET. Different proxies may cache the results or the browser may open the link automatically without clicking (prefetching). Either use PUT or POST.

Reply:
Because I can't change the request method of an `<a>` tag that means I have to use a form instead which only allows the `POST` method so I'll go with that.

Reply:
Yeah, there's no nice way of doing it without form. But the form one isn't too complicated:

````
<form method="DELETE" action="/foo/{{ .ID}}">
    <input type="submit" value="do it">
</form>
````

And you can use post as well:

````
<form method="POST" action="/stop">
    <input type="hidden" id="offer-id" value="{{.ID}}">
    <input type="submit" value="do it">
</form>
````

There are also hacks that use onclick + javascript to attach itself to `a` tag. And I suspect they might have bizarre problems on certain devices.